src/dram_controller.cc

```diff
@@ -554,19 +554,55 @@ bool MEMORY_CONTROLLER::add_wq(const request_type& packet)
 
 unsigned long DRAM_ADDRESS_MAPPING::get_channel(champsim::address address) const
 {
-  return std::get<SLICER_CHANNEL_IDX>(address_slicer(address)).to<unsigned long>();
+  unsigned long channel = std::get<SLICER_CHANNEL_IDX>(address_slicer(address)).to<unsigned long>();
+  champsim::address row = champsim::address{get_row(address)};
+
+  //channel bits should be xor'd with each row bit
+  unsigned long c_bits = champsim::size(get<SLICER_CHANNEL_IDX>(address_slicer));
+  unsigned long r_bits = champsim::size(get<SLICER_ROW_IDX>(address_slicer));
+  for(unsigned long r = 0; r < r_bits; r++) {
+    for(unsigned long c = 0; c < c_bits; c++)
+      channel ^= row.slice_lower<champsim::data::bits{0}>().to<unsigned long>() << c;
+    row = champsim::address{row.to<unsigned long>() >> 1};
+  }
```

Comment:
Similar code shows up 3 times here. Is there a way to extract this into a `DRAM_ADDRESS_MAPPING::swizzle_bits(...)` function?